core/src/main/java/discord4j/core/object/entity/ApplicationEmoji.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * This file is part of Discord4J.
+ *
+ * Discord4J is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Discord4J.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package discord4j.core.object.entity;
+
+import discord4j.common.util.Snowflake;
+import discord4j.core.GatewayDiscordClient;
+import discord4j.core.spec.ApplicationEmojiEditMono;
+import discord4j.core.spec.ApplicationEmojiEditSpec;
+import discord4j.core.util.EntityUtil;
+import discord4j.discordjson.json.EmojiData;
+import discord4j.discordjson.json.UserData;
+import reactor.core.publisher.Mono;
+import reactor.util.annotation.Nullable;
+
+import java.util.Objects;
+
+/**
+ * A Discord application emoji.
+ * <br>
+ * <a href="https://discord.com/developers/docs/resources/emoji#emoji-resource">Emoji Resource</a>
+ */
+public final class ApplicationEmoji extends Emoji {
+
+    /** The ID of the application this emoji is associated to. */
+    private final long applicationId;
+
+    /**
+     * Constructs a {@code ApplicationEmoji} with an associated {@link GatewayDiscordClient} and Discord data.
+     *
+     * @param gateway The {@link GatewayDiscordClient} associated to this object, must be non-null.
+     * @param data The raw data as represented by Discord, must be non-null.
+     * @param applicationId The ID of the application this emoji is associated to.
+     */
+    public ApplicationEmoji(final GatewayDiscordClient gateway, final EmojiData data, final long applicationId) {
+        super(gateway, data);
+        this.applicationId = applicationId;
+    }
+
+    /**
+     * Gets the ID of the application this emoji is associated to.
+     *
+     * @return The ID of the application this emoji is associated to.
+     */
+    public Snowflake getApplicationId() {
+        return Snowflake.of(applicationId);
+    }
+
+    /**
+     * Requests to retrieve the application this emoji is associated to.
+     *
+     * @return A {@link Mono} where, upon successful completion, emits the {@link ApplicationInfo application} this emoji is associated
+     * to. If an error is received, it is emitted through the {@code Mono}.
+     */
+    public Mono<ApplicationInfo> getApplication() {
```

Comment:
The emoji might not be associated with the current application if it is read from a message sent by an app. (I don't know how these would be parsed tho)

Reply:
Oh, okay then 👌